// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "calculator";
option java_outer_classname = "Calculatorproto";
option objc_class_prefix = "HLW";

package calculator;

// The Calculator service definition.
service Calculator {
  // Sends a greeting
  rpc Add (RequestAdd) returns (RequestReply) {}
}

// The request message containing the two numbers
message RequestAdd {
  double inputA=1;
  double inputB=2;
}
// The response containing the result
message RequestReply {
  double result=1;
}